### PR TITLE
fix updating suspended component

### PR DESCRIPTION
**Pull request by tanhauhau**
replace the `parentDom` of a suspended component to a detached `<div />`, allowing `diff` to continue updating the component without affecting the actual dom.

when updating a suspended children that will no longer suspending, allow `diff` to insert elements to the detached `<div />`, and moved it over to original `parentDom` if the `<Suspense />` unsuspense.

this change allow to re-enable some of the test cases, but i'm not sure whether this is the right approach on this, probably will add a few more tests, but hope to know whether im approach it in the right direction.

### CODE REVIEW

**Comment by tanhauhau** [comment]
fyi @kenrick95

**Review by marvinhagemeister — APPROVED** [review]
Woa, this is outstanding! Thank you so much for the PR 🙌 🚀

File: compat/test/browser/suspense.test.js [hunk 1]
```diff
@@ -1706,25 +1761,141 @@ describe('suspense', () => {
...
+					this.setState({ promise: null, content });
+				}
+
+				render() {
+					if (this.state.promise) {
+						throw this.state.promise;
+					}
+
+					return this.state.content;
+				}
```
Comment (andrewiggins) [inline compat/test/browser/suspense.test.js:1790]: Haven't seen this pattern before. Is this something you do in your own apps or just a quick way to write a test? Specifically referring to throwing a promise that never resolves. Would just like to better understand this use case 😅

**Review by andrewiggins — COMMENTED** [review]
@tanhauhau Thanks again for your PR :) Sorry for the belated review, but just had one question about better understanding how you use this new functionality

File: compat/test/browser/suspense.test.js (hunk 1 above)
Comment (tanhauhau) [reply compat/test/browser/suspense.test.js:1790]: Well, it was copied from the test case "should allow suspended children to update", which was written by you 🤣

I believe it's just a quick wat to write the test.

Comment (andrewiggins) [reply compat/test/browser/suspense.test.js:1790]: OH! Ahahahaha 😂😂😊 Got it, thanks :)

I'm thinking we may need to re-evaluate if this use case is really necessary at this moment (unsuspending without resolving the Promise). [React doesn't support it](https://codesandbox.io/s/resume-with-rerender-kim18) and currently it requires that we override the render method of a developer's component which feels icky. I know we do this for `componentWillUnmount` already, but its not a great pattern and one we should move away from (an `options.unmount` plugin would be better). We'll likely revisit this later. Feel free to leave your thoughts on PR #2917 😄

Comment (tanhauhau) [reply compat/test/browser/suspense.test.js:1790]: Hmmm.. I wonder will adding options.commit and options.unmount be in the plan for Preact X, i'd love to help adding them in

**Pull request merged**